gtfmerge/genome1.cc

```diff
@@ -147,6 +150,21 @@ int genome1::write(const string &file)
 	return 0;
 }
 
+int genome1::write_samples(const string &file)
+{
+	string samples_file = file + ".samples.txt";
+	ofstream fout(samples_file);
+	if(fout.fail()) return 1;
+	for(int i = 0; i < transcripts.size(); i++)
+	{
+		transcript &t = transcripts[i];
+		if(t.coverage < min_transcript_coverage) continue;
+		fout << t.gene_id << "\t" << t.samples << endl;
```

Comment:
Do not use `endl`, it will slow down things tremendously. Use `'\n`.

(You may already know that, but `endl` appends a new line and does a `flush` on the stream. It is good for things that need to be visible immediately, such as logging or error messages. For writing bulk data, `endl` removes any benefit of caching and is much much slower).